### PR TITLE
backport to 1.18.2

### PR DESCRIPTION
**Pull request by CCr4ft3r**
I somehow missed the plan to backport it ...
But now it's done and ready to release :)

You can create a new 1.18.2 branch based on the 1.19.2 one and change the target branch to it after reviewing the changes.
However there are still some whitespaces but this time that's not an issue being caused by IntelliJ formatter since these weren't visible before pushing it. I think that's the diff viewer of GitHub itself. You can hide whitespace changes for the review: https://github.blog/2018-05-01-ignore-white-space-in-code-review/

### CODE REVIEW

**Comment by CCr4ft3r** [comment]
The feature hide whitespaces seems to be buggy on GitHub. However comparing it locally in IntelliJ can avoid the need of comparing blanks

File: fabric/src/main/java/mod/crend/autohud/fabric/compat/mixin/overflowingbars/ArmorBarRendererMixin.java
Comment (Crendgrim) [inline fabric/src/main/java/mod/crend/autohud/fabric/compat/mixin/overflowingbars/ArmorBarRendererMixin.java:1]: Fully remove this file please

**Comment by Crendgrim** [comment]
Thank you! I'll do some tests on the changes and then publish it, hopefully this weekend.

File: fabric/src/main/java/mod/crend/autohud/fabric/compat/mixin/overflowingbars/ArmorBarRendererMixin.java
Comment (CCr4ft3r) [reply fabric/src/main/java/mod/crend/autohud/fabric/compat/mixin/overflowingbars/ArmorBarRendererMixin.java:1]: Done

**Comment by Crendgrim** [comment]
Fabric version doesn't launch for me in dev environment, but seems to work when compiled and loaded. Probably some old toolchain issue. Quick tests seem to suggest everything works well, good job! Thanks for the backport.

**Comment by Crendgrim** [comment]
Published, and even managed to properly handle the GitHub side this time :tada:

**Comment by CCr4ft3r** [comment]
> Fabric version doesn't launch for me in dev environment, but seems to work when compiled and loaded. Probably some old toolchain issue. Quick tests seem to suggest everything works well, good job! Thanks for the backport.

I also noticed that problem and then I tested it outside. 
No problem :) It's a great mod.

**Comment by CCr4ft3r** [comment]
In my tests the hotbar was being hided correctly after some seconds.

**Comment by CCr4ft3r** [comment]
Seems to be a strange problem with Forge versions prior to 40.2.4. On other versions, the HUD is being hided as expected.
However, in all these versions all mixins could be applied without warnings or errors.
So the best would be to define the forge version 40.2.4 as minimum in mods.toml.

**Comment by CCr4ft3r** [comment]
The user who reported the issue on CurseForge was using Forge 40.2.10. But with a clean setup it's working fine for me.
Tested 40.2.9 and 40.2.10

**Comment by Crendgrim** [comment]
Okay, I officially am confused. Tested it again today to try figure it out, and it just... works. 😕

**Comment by CCr4ft3r** [comment]
🤣 For versions of 40.2.4 and above I tested it multiple times and it worked always fine. For 40.2.2 it failed immediately. 
So is 40.2.2 working fine for you?

I currently don't think that this is like a race condition of multiple mixins since no warnings or erros were logged in the debug.log when hiding of the HUD didn't worked.

**Pull request merged**